modules/bpi/bpi.images.inc

```diff
@@ -213,3 +229,71 @@ function bpi_fetch_image($filename, $image_url) {
 
   return file_save_data($result->data, $filename, FILE_EXISTS_REPLACE);
 }
+
+/**
+ * Compute the image type from a field name on a node.
+ *
+ * @param string $image_field_name
+ *   The image field name.
+ * @param array $node
+ *   The node.
+ *
+ * @return NULL|string
+ *   The image type.
+ */
+function bpi_get_image_type($image_field_name, array $node) {
+  /*
+   * We assume that the image type is the last part of the field name, i.e.
+   *   field_{{node type}}_{{image type}}
+   */
+  $pattern = '/field_' . preg_quote($node->type, '/') . '_(?<image_type>.+)/';
+  if (preg_match($pattern, $image_field_name, $matches)) {
+    return $matches['image_type'];
+  }
+  return NULL;
+}
+
+/**
+ * Store syndicated images in session.
+ *
+ * @param mixed $syndicated_images
+ *   The value to store.
+ */
+function bpi_set_syndicated_images($syndicated_images) {
+  $_SESSION['bpi']['syndicated_images'] = $syndicated_images;
+}
+
+/**
+ * Get syndicated images from session.
+ *
+ * @return mixed
+ *   The value if any.
+ */
+function bpi_get_syndicated_images() {
+  return isset($_SESSION['bpi']['syndicated_images']) ? $_SESSION['bpi']['syndicated_images'] : NULL;
```

Comment:
I use NULL to indicate "no value". This is used by the "fishy" part of the code (https://github.com/ding2/ding2/pull/55/files#r46263721).